### ISSUE TITLE
Problems with the "unloading document cleanup steps"

### ISSUE THREAD

**Issue by foolip** [whatwg/fullscreen#82]
The spec currently says says "Whenever the unloading document cleanup steps run with a document, fully exit fullscreen document."

Working on https://github.com/whatwg/fullscreen/pull/65, I wrote https://github.com/w3c/web-platform-tests/pull/5887 which was supposed to test this and passes in Firefox, but I've realized there are at least these problems:
* The exit fullscreen algorithm does nothing if the document is not fully active. Is a document fully active when [unload a document](https://html.spec.whatwg.org/multipage/browsers.html#unload-a-document) runs? @annevk?
* If the exit fullscreen algorithm did run as usual, part of it is async after the resize, and event should then be fired. They aren't in Firefox, and I don't think it'd make sense to fire events.

It's tempting to just use [unfullscreen a document](https://fullscreen.spec.whatwg.org/#unfullscreen-a-document), but when the top-level browsing context is navigated *something* has to "resize topLevelDoc’s viewport to its "normal" dimensions" and that's currently a part of the exit fullscreen algorithm.

**Comment by annevk** [comment]
I think it's still active while unload a document runs, yes, but that's also one of the less clear areas of the HTML specification I hope to refactor.

**Comment by foolip** [comment]
OK, that isn't totally surprising, but I assume that it will become not active shortly after and that at least firing events on it isn't actually something we want. (Even more weird would be if the events are queued and then arrive if the document becomes active again.)

**Comment by annevk** [comment]
Yeah, if we needed to dispatch any events there they'd have to be carefully vetted I think. Even though fullscreen is a special state, it still shouldn't make it harder to navigate and such I think.

**Comment by foolip** [comment]
There are some tests to revive in https://github.com/web-platform-tests/wpt/pull/5887 if I get back to fixing this spec issue.
